{"kind":"symbol","identifier":{"interfaceLanguage":"swift","url":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError"},"topicSections":[{"generated":true,"title":"Initializers","anchor":"Initializers","identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError\/init(_:)"]},{"anchor":"Instance-Properties","identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError\/description","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError\/message"],"generated":true,"title":"Instance Properties"},{"identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError\/Error-Implementations"],"generated":true,"anchor":"Default-Implementations","title":"Default Implementations"}],"metadata":{"symbolKind":"struct","fragments":[{"kind":"keyword","text":"struct"},{"text":" ","kind":"text"},{"text":"ValidationError","kind":"identifier"}],"navigatorTitle":[{"kind":"identifier","text":"ValidationError"}],"modules":[{"name":"ArgumentParser"}],"role":"symbol","title":"ValidationError","roleHeading":"Structure","externalID":"s:14ArgumentParser15ValidationErrorV"},"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/argumentparser\/validationerror"]}],"schemaVersion":{"minor":3,"major":0,"patch":0},"sections":[],"abstract":[{"text":"An error type that is presented to the user as an error with parsing their","type":"text"},{"text":" ","type":"text"},{"type":"text","text":"command-line input."}],"relationshipsSections":[{"kind":"relationships","title":"Conforms To","type":"conformsTo","identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/s23CustomStringConvertibleP","doc:\/\/swift-argument-parser.ArgumentParser\/s5ErrorP","doc:\/\/swift-argument-parser.ArgumentParser\/s8SendableP"]}],"seeAlsoSections":[{"generated":true,"identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/Validation","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/CleanExit","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode"],"title":"Validation and Errors","anchor":"Validation-and-Errors"}],"hierarchy":{"paths":[["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser"]]},"primaryContentSections":[{"declarations":[{"platforms":["macOS"],"tokens":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"ValidationError","kind":"identifier"}],"languages":["swift"]}],"kind":"declarations"}],"references":{"doc://swift-argument-parser.ArgumentParser/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/s8SendableP"},"doc://swift-argument-parser.ArgumentParser/s23CustomStringConvertibleP":{"type":"unresolvable","title":"Swift.CustomStringConvertible","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/s23CustomStringConvertibleP"},"doc://swift-argument-parser.ArgumentParser/s5ErrorP":{"type":"unresolvable","title":"Swift.Error","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/s5ErrorP"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/CleanExit":{"navigatorTitle":[{"text":"CleanExit","kind":"identifier"}],"abstract":[{"type":"text","text":"An error type that represents a clean (i.e. non-error state) exit of the"},{"text":" ","type":"text"},{"type":"text","text":"utility."}],"type":"topic","title":"CleanExit","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/CleanExit","url":"\/documentation\/argumentparser\/cleanexit","kind":"symbol","role":"symbol","fragments":[{"kind":"keyword","text":"struct"},{"text":" ","kind":"text"},{"text":"CleanExit","kind":"identifier"}]},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode":{"abstract":[{"text":"An error type that only includes an exit code.","type":"text"}],"type":"topic","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode","title":"ExitCode","role":"symbol","url":"\/documentation\/argumentparser\/exitcode","navigatorTitle":[{"kind":"identifier","text":"ExitCode"}],"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"text":"ExitCode","kind":"identifier"}],"kind":"symbol"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/Validation":{"kind":"article","title":"Providing Custom Validation","url":"\/documentation\/argumentparser\/validation","abstract":[{"type":"text","text":"Provide helpful feedback to users when things go wrong."}],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/Validation","role":"article","type":"topic"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ValidationError":{"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError","type":"topic","navigatorTitle":[{"kind":"identifier","text":"ValidationError"}],"fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"ValidationError","kind":"identifier"}],"url":"\/documentation\/argumentparser\/validationerror","kind":"symbol","role":"symbol","abstract":[{"type":"text","text":"An error type that is presented to the user as an error with parsing their"},{"text":" ","type":"text"},{"type":"text","text":"command-line input."}],"title":"ValidationError"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ValidationError/Error-Implementations":{"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError\/Error-Implementations","kind":"article","abstract":[],"role":"collectionGroup","title":"Error Implementations","url":"\/documentation\/argumentparser\/validationerror\/error-implementations","type":"topic"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser":{"title":"ArgumentParser","type":"topic","abstract":[{"text":"Straightforward, type-safe argument parsing for Swift.","type":"text"}],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser","role":"collection","url":"\/documentation\/argumentparser","kind":"symbol"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ValidationError/description":{"kind":"symbol","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError\/description","role":"symbol","url":"\/documentation\/argumentparser\/validationerror\/description","type":"topic","abstract":[],"fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"description","kind":"identifier"},{"text":": ","kind":"text"},{"text":"String","preciseIdentifier":"s:SS","kind":"typeIdentifier"}],"title":"description"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ValidationError/init(_:)":{"title":"init(_:)","type":"topic","abstract":[{"type":"text","text":"Creates a new validation error with the given message."}],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError\/init(_:)","role":"symbol","url":"\/documentation\/argumentparser\/validationerror\/init(_:)","fragments":[{"text":"init","kind":"identifier"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":")","kind":"text"}],"kind":"symbol"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ValidationError/message":{"type":"topic","kind":"symbol","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError\/message","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"message"},{"text":": ","kind":"text"},{"text":"String","preciseIdentifier":"s:SS","kind":"typeIdentifier"}],"url":"\/documentation\/argumentparser\/validationerror\/message","title":"message","abstract":[{"text":"The error message represented by this instance, this string is presented to","type":"text"},{"text":" ","type":"text"},{"text":"the user when a ","type":"text"},{"type":"codeVoice","code":"ValidationError"},{"type":"text","text":" is thrown from either; "},{"type":"codeVoice","code":"run()"},{"type":"text","text":","},{"type":"text","text":" "},{"code":"validate()","type":"codeVoice"},{"text":" or a transform closure.","type":"text"}],"role":"symbol"}}}